zulip_bots/zulip_bots/bots/beeminder/beeminder.py

```diff
@@ -0,0 +1,80 @@
+import requests
+import logging
+import json
+from typing import Dict, Any, List
+
+help_message = '''Beeminder bot.\
+Simply send this bot a message following the syntax shown below,\
+and it will respond depending on the configured bot's behaviour.\
+\nSYNTAX - @mention-botname daystamp  value  comment (NOTE: 2 spaces between inputs)\
+\n* daystamp -> Enter date in this manner: dd [NOTE: optional field, default -> current daystamp],\
+\n* value -> Enter a value,\
+\n* comment -> Add a comment [NOTE: optional field, default -> None]\
+'''
+
+def get_beeminder_response(message_content: str, config_info: Dict[str, str]) -> str:
+    username = config_info['username']
+    goalname = config_info['goalname']
+    key = config_info['key']
+
+    if message_content.strip() == '' or message_content.strip() == 'help':
+        return help_message
+
+    url = "https://www.beeminder.com/api/v1/users/{}/goals/{}/datapoints.json?auth_token={}".format(username, goalname, key)
+    message_pieces = message_content.split('  ', 3)
+
+    # for len(message_pieces)==2 we check for last value type in list to know whether user entered daystamp & value or value & comment
```

Comment:
Consider wrapping long lines. It won't be pleasant for people with small monitors.

Reply:
@ViRu-ThE-ViRuS  Okay i'll do it...but my travis build is failing, having assertion error (401 unauthorized )..i doubt do i need to provide my original key and info in `.conf` file ?

Reply:
You need to use fixtures here. Checkout out the `giphy` bot for example implementation.

Reply:
Okay got it! I'll provide a fixture for that. Thanx for reviewing this :)